keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/multiple/LambdaModelVertexTest.java

```diff
@@ -0,0 +1,274 @@
+package io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.multiple;
+
+import io.improbable.keanu.algorithms.NetworkSamples;
+import io.improbable.keanu.algorithms.mcmc.MetropolisHastings;
+import io.improbable.keanu.algorithms.variational.optimizer.nongradient.NonGradientOptimizer;
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.tensor.Tensor;
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.VertexLabel;
+import io.improbable.keanu.vertices.bool.BoolVertex;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import io.improbable.keanu.vertices.intgr.IntegerVertex;
+import io.improbable.keanu.vertices.model.LambdaModelVertex;
+import io.improbable.keanu.vertices.model.ModelVertex;
+import org.bytedeco.javacpp.annotation.Const;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class LambdaModelVertexTest {
+
+    /*
+    The model we are mimicking here is a Java function, modelExecution.
+
+    It takes one input, Temperature, and produces two outputs of type double, Chance of Rain & Humidity. These outputs
+    are written to file.
+
+    It also produces one integer output and one boolean output. Suggested Factor of Suncream and 'is it sunny'. These
+    are also written to file.
+     */
+
+    @Mock
+    private BufferedReader rainReader;
+
+    @Mock
+    private BufferedReader humidityReader;
+
+    @Mock
+    private BufferedReader suggestedFactorSuncream;
+
+    @Mock
+    private BufferedReader isSunnyReader;
+
+    private KeanuRandom random;
+    private DoubleVertex inputToModel;
+
+    @Before
+    public void setup() throws IOException {
+        random = new KeanuRandom(1);
+        rainReader = mock(BufferedReader.class);
+        humidityReader = mock(BufferedReader.class);
+        suggestedFactorSuncream = mock(BufferedReader.class);
+        isSunnyReader = mock(BufferedReader.class);
+
+        when(rainReader.readLine()).thenAnswer(new Answer<Object>() {
```

Comment:
If you don't need the arguments to a mocked method then I think you can do just `.thenReturn()`